pkg/scheduler/framework/plugins/nodeports/node_ports.go

```diff
@@ -97,6 +98,15 @@ func getPreFilterState(cycleState *framework.CycleState) (preFilterState, error)
 	return s, nil
 }
 
+// EventsToRegister returns the possible events that may make a Pod
+// failed by this plugin schedulable.
+func (pl *NodePorts) EventsToRegister() []framework.ClusterEvent {
+	return []framework.ClusterEvent{
+		{Resource: framework.Pod, ActionType: framework.Delete},
```

Comment:
LGTM.

BTW: let's add a comment to explain why PodUpdate doesn't need to be registered:

```
Due of immutable fields `spec.containers[*].ports`, PodUpdate event is ignored.
```

Reply:
@Huang-Wei Added. Thanks for your suggestions and help.